lib/record_store.rb

```diff
@@ -31,6 +31,8 @@
 require 'record_store/provider/dynect'
 require 'record_store/provider/dnsimple'
 require 'record_store/provider/google_cloud_dns'
+require 'record_store/provider/ns1/client'
```

Comment:
Do we need to require both of them from here, or is the client only required by the provider ? (in which case, we should only require the provider here, and require the client from the provider only)

Reply:
I might be wrong but I believe we need both of them from here.

Reply:
If you require `record_store/provider/ns1` and `record_store/provider/ns1.rb` requires `record_store/provider/ns1/client`, then the requirement is already there, just indirect.
It "hides" the fact that that "client" level exists, as the `record_store.rb` file does not care about it. This works mostly because `record_store.rb` uses nothing that's in `ns1/client.rb` but only things in `ns1.rb`. The one using `ns1/client.rb` is `ns1.rb`.

If A requires B and B requires C, indirectly, A will require C, so we're still good, but it's cleaner :)

Reply:
Great! Thanks for the details! Yeah, it all makes sense, much cleaner now.